test/Serialization/modularization-error.swift

```diff
@@ -0,0 +1,50 @@
+/// Simulate typical modularization issues using Swift modules.
+// RUN: %empty-directory(%t)
+// RUN: split-file %s %t
+
+/// Compile two library modules A and B, and a client.
+// RUN: %target-swift-frontend %t/LibOriginal.swift -emit-module-path %t/A.swiftmodule -module-name A
+// RUN: %target-swift-frontend %t/Empty.swift -emit-module-path %t/B.swiftmodule -module-name B
+// RUN: %target-swift-frontend %t/Client.swift -emit-module-path %t/Client.swiftmodule -module-name Client -I %t
+
+/// Move MyType from A to B.
+// RUN: %target-swift-frontend %t/Empty.swift -emit-module-path %t/A.swiftmodule -module-name A
+// RUN: %target-swift-frontend %t/LibOriginal.swift -emit-module-path %t/B.swiftmodule -module-name B
+// RUN: not %target-swift-frontend -emit-sil %t/Client.swiftmodule -module-name Client -I %t 2>&1 \
+// RUN:   | %FileCheck --check-prefixes CHECK,CHECK-MOVED %s
+// CHECK-MOVED: <unknown>:0: error: module reference to top-level 'MyType' broken by a context change, was found in 'A' when building 'Client', but now a candidate is found only in 'B'. Audit related headers or uninstall SDK roots.
+
+/// Change MyType into a function.
+// RUN: %target-swift-frontend %t/LibTypeChanged.swift -emit-module-path %t/A.swiftmodule -module-name A
+// RUN: %target-swift-frontend %t/Empty.swift -emit-module-path %t/B.swiftmodule -module-name B
+// RUN: not %target-swift-frontend -emit-sil %t/Client.swiftmodule -module-name Client -I %t 2>&1 \
+// RUN:   | %FileCheck --check-prefixes CHECK,CHECK-TYPE-CHANGED %s
+// CHECK-TYPE-CHANGED: <unknown>:0: error: module reference to top-level 'MyType' broken by a context change, its type changed since building 'Client', it was in 'A' now found in 'A'. Ensure Swift language version match between modules or uninstall SDK roots.
```

Comment:
Is it possible to show the type info here? e.g. it was a struct but now a func in module 'A'

Reply:
I should be able to. Since it failed to deserialize we pretty much just know if it's a type or if it's a ValueDecl. In most cases it's a type so having that precision should help.